bridge/src/migration.rs

```diff
@@ -0,0 +1,68 @@
+#[allow(unused_imports)]
+use super::*;
+
+#[cfg(feature = "try-runtime")]
+use frame_support::ensure;
+#[cfg(feature = "try-runtime")]
+use sp_std::vec::Vec;
+use frame_support::traits::{Get, OnRuntimeUpgrade, StorageVersion};
+use log;
+use sygma_traits::MpcAddress;
+
+const EXPECTED_STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+#[cfg(feature = "try-runtime")]
+const FINAL_STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+const MPC_ADDR: &str = "B01137123EF02fAeF251a39108c6ef513AAaC485";
+
+pub struct FixMpcAddress<T>(sp_std::marker::PhantomData<T>);
+
+impl<T: Config> OnRuntimeUpgrade for FixMpcAddress<T> {
+	fn on_runtime_upgrade() -> frame_support::weights::Weight {
```

Comment:
Few questions here:

1. if I understand correctly, `on_runtime_upgrade`, `pre_upgrade` and `post_upgrade` func will be triggered when doing runtime upgrade, like call `tx.parachainSystem.authorizedUpgrade`? 

2. this migration script only migrate MPC address currently, but technically it should be able to update any thing in the storage, correct?

3. if we migrate MPC address to this new address this time, next time if we do another runtime upgrade but we don't want to migrate MPC address, will this script get triggered? so basically I want to ask, does this migrate script gets run every time we do a runtime upgrade? I'm thinking that if there are old value in this script but we accidentally migrate them back to old values

Reply:
1. only `on_runtime_upgrade` will be executed on real runtime upgrade. `pre_upgrade` and `post_upgrade` only be executed when will execute `try-runtime` which is a test suit that allows us test the migration script based on production on-chain data before we push it to production in case we do something unexpected thing that corrupt the on-chain data

2. Technically yes

3. We need to remove the script from runtime (generally we will remove the script too), e.g. remove it from [here](https://github.com/sygmaprotocol/sygma-substrate-pallets/blob/2ac9e9da4b2a3a77b2c0502dccd49e5aadb1ac57/substrate-node/runtime/src/lib.rs#L781) in our runtime. And definitely, if we don't remove it then will be executed every time when upgrade the runtime. We should never make this happen

Reply:
ok got it, thanks for the answer!